main.tf

```
@@ -36,3 +42,30 @@ resource "aws_security_group" "elasticache" {
     protocol    = "tcp"
   }
 }
+
+resource "aws_elasticache_replication_group" "default" {
+  count = var.replication_group ? 1 : 0
+
+  at_rest_encryption_enabled    = var.at_rest_encryption
+  auth_token                    = var.in_transit_encryption ? var.auth_token : null
```

This can just be like this if the default is null:
```suggestion
  auth_token                    = var.auth_token
```